app/controllers/Contributions.java

```diff
@@ -3338,6 +3340,76 @@ public static Result createAnonymousContributionOnAssembly(@ApiParam(name = "uui
             return ok(Json.toJson(c));
         }
     }
+
+    /**
+     * GET       /api/assembly/:aid/campaign/:cid/contribution/:coid/body
+     *
+     * @param aid
+     * @param cid
+     * @param coid
+     * @return
+     */
+    @ApiOperation(httpMethod = "GET", response = String.class, produces = "application/json", value = "Get the pad body url of a Contribution in a Assembly")
+    @ApiResponses(value = {@ApiResponse(code = 404, message = "No contributions found", response = TransferResponseStatus.class)})
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "SESSION_KEY", value = "User's session authentication key", dataType = "String", paramType = "header")})
+    public static Result findAssemblyContributionPadBody(
+            @ApiParam(name = "aid", value = "Assembly ID") Long aid,
+            @ApiParam(name = "cid", value = "Campaign ID") Long cid,
+            @ApiParam(name = "coid", value = "Contribution ID") Long coid,
+            @ApiParam(name = "rev", value = "Integer", defaultValue = "0") Integer rev,
+            @ApiParam(name = "format", value = "String", allowableValues = "text, html", defaultValue = "html") String format) {
+        Contribution c = Contribution.read(coid);
+        String etherpadServerUrl = Play.application().configuration().getString(GlobalData.CONFIG_APPCIVIST_ETHERPAD_SERVER);
+        String etherpadApiKey = Play.application().configuration().getString(GlobalData.CONFIG_APPCIVIST_ETHERPAD_API_KEY);
+        if (c != null) {
+            Integer revision = rev !=null && rev != 0 ? rev : c.getPublicRevision();
+            Resource pad = c.getExtendedTextPad();
+            String padId = pad.getPadId();
+            String finalFormat = format != null && format == "text" ? "TEXT":"HTML";
+            if (padId != null) {
+                String url = etherpadServerUrl+"/api/1/get"+finalFormat+"?padID="+padId+"&rev="+revision+"&apikey="+etherpadApiKey;
+
+                return ok(Json.toJson(url));
```

Comment:
I think I was not clear with this: what we need to return is the result to that Etherpad API call. 
BTW, there is a `utils.services.EtherpadWrapper` class where you can include your implementation of this endpoint request.